canaries/2D_image_stream.py

```diff
@@ -56,22 +54,23 @@ def get_frames_per_sec(ordered_frames):
         ordered_frames.popleft()
     return len(ordered_frames)
 
-async def detection_stream(robot, llist2):
+
+async def detection_stream(robot):
+    llist = deque()
     transform_cam_name = "transform"
     transform = Camera.from_robot(robot, transform_cam_name)
     logging.info(f"found camera {transform_cam_name}")
     while True:
         # This is to stop this script just before the start of the next hour.
         current_min = int(datetime.datetime.now().strftime("%M"))
         if current_min == 59:
-            await close_robot(robot)
```

Comment:
This fixes the bugs we were seeing. Specifically this solves two problems:
1. Either the `detection_stream` or `image_stream` will call `close_robot` first causing the other function to error with the message `Connection lost` and possibly freeze.
2. Because 1 throws an exception, it's caught by our error handler which exits with exit code of 1. This causes the update script to re-launch this.